2nd-gen/packages/core/components/asset/Asset.base.ts

```diff
@@ -10,14 +10,67 @@
  * governing permissions and limitations under the License.
  */
 
+import { PropertyValues } from 'lit';
 import { property } from 'lit/decorators.js';
 
 import { SpectrumElement } from '@spectrum-web-components/core/shared/base/index.js';
 
+import { ASSET_VARIANTS, type AssetVariant } from './Asset.types.js';
+
+/**
+ * @slot - The content to render when no `variant` is provided (typically an <img> element)
+ */
 export abstract class AssetBase extends SpectrumElement {
+    // ─────────────────────────
+    //     API TO OVERRIDE
+    // ─────────────────────────
+
+    /**
+     * @internal
+     *
+     * A readonly array of all valid variants for the asset.
+     *
+     * This is an actual internal property, intended not for customer use
+     */
+    static readonly VARIANTS: readonly AssetVariant[] = ASSET_VARIANTS;
+
+    // ─────────────────
+    //     SHARED API
+    // ─────────────────
+
+    /**
+     * The variant of the asset. When not provided, slot content is rendered (e.g., an image).
+     */
     @property({ type: String, reflect: true })
-    public variant: 'file' | 'folder' | undefined;
+    public variant: AssetVariant | undefined;
 
+    /**
+     * Accessible label for the asset’s SVG variant.
```

Comment:
Super nit pick: our variants are technically named them "file" or "folder," and not "SVG." Is it odd or confusing to mention an SVG variant? I'll leave it up to you if you want to change it.

```suggestion
* Accessible label for the asset’s file and folder variants.
```